File: app/src/main/java/com/halfplatepoha/jisho/history/HistoryFragment.java
package com.halfplatepoha.jisho.history;

import android.support.annotation.Nullable;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.halfplatepoha.jisho.R;
import com.halfplatepoha.jisho.base.BaseFragment;
import com.halfplatepoha.jisho.db.History;

import butterknife.BindView;
import io.realm.Realm;

public class HistoryFragment extends BaseFragment<HistoryContract.Presenter> implements HistoryAdapter.OnSearchClickedListener, HistoryContract.View{

    @BindView(R.id.rlHistory)
    RecyclerView rlHistory;

    HistoryAdapter adapter;

    Realm realm;

    HistoryFragmentActionListener listener;

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        adapter = new HistoryAdapter(getActivity());
        adapter.setSearchClickedListener(this);
        rlHistory.setAdapter(adapter);
        rlHistory.setLayoutManager(new LinearLayoutManager(getActivity(), LinearLayoutManager.VERTICAL, true));

        refreshUI();
    }

    public void setHistoryFragmentActionListener(HistoryFragmentActionListener listener) {
        this.listener = listener;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        realm = Realm.getInstance(getApplication().getAppConfig());
    }

    private void refreshUI() {
        History history = realm.where(History.class).findFirst();
        if(history != null && history.getHistory() != null) {
            for(String str : history.getHistory())
                adapter.addHistory(str);
        }
    }

    @Override
    public void onSearchClicked(String searchString) {
        if(listener != null)
            listener.onSearchHistoryStringSelected(searchString);
    }

    @Override
    public int getLayoutRes() {
        return R.layout.fragment_history;
    }

    public interface HistoryFragmentActionListener {
        void onSearchHistoryStringSelected(String searchString);
    }
}

File: app/src/main/java/com/halfplatepoha/jisho/v2/history/HistoryFragment.kt
package com.halfplatepoha.jisho.v2.history

import android.os.Bundle
import android.view.View
import androidx.lifecycle.Observer
import com.halfplatepoha.jisho.R
import com.halfplatepoha.jisho.v2.base.BaseFragment
import com.halfplatepoha.jisho.v2.detail.DetailActivity
import com.halfplatepoha.jisho.v2.viewmodel.HistoryViewModel
import org.koin.android.scope.currentScope
import org.koin.android.viewmodel.ext.android.viewModel

class HistoryFragment : BaseFragment() {

    val historyFragmentViewModel: HistoryFragmentViewModel by currentScope.viewModel(this)

    val historyViewModel: HistoryViewModel by viewModel()

    val historyAdapter: HistoryAdapter by currentScope.inject()

    override val layoutId: Int
        get() = R.layout.fragment_history

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        historyViewModel.historyItems.observe(this, Observer { historyAdapter.historyItems = it })
    }

    override fun setListeners() {
        historyAdapter.listener = { historyFragmentViewModel.onHistorySelected(it) }
    }

    override fun setViewModelObservers() {
        historyFragmentViewModel.selectedHistoryItem.observe(this, Observer {
            startActivity(DetailActivity.getLaunchIntent(context!!, it))
        })
    }

}